src/Bicep.Core/Registry/Oci/OciArtifactResult.cs

```diff
@@ -14,12 +14,12 @@ namespace Bicep.Core.Registry.Oci
 {
     public class OciArtifactResult
     {
-        public OciArtifactResult(BinaryData manifestBits, string manifestDigest, ImmutableArray<(string MediaType, BinaryData Data)> layers)
+        public OciArtifactResult(BinaryData manifestBits, string manifestDigest, ImmutableDictionary<string, (string MediaType, BinaryData Data)> layers)
         {
             this.manifestBits = manifestBits;
             this.Manifest = OciManifest.FromBinaryData(manifestBits) ?? throw new InvalidOperationException("the manifest is not a valid OCI manifest");
             this.ManifestDigest = manifestDigest;
-            this.Layers = layers;
+            this.Layers = layers.Values;
```

Comment:
Why change the `layers` argument to a dictionary if it's not used as such?

Reply:
yeah, I thought about it and a record seems a better approach